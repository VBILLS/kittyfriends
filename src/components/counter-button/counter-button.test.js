import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import CounterButton from './counter-button.component';

describe('CounterButton component', () => {
  const mockColor = 'purple';
  const wrapper = shallow(<CounterButton color={mockColor} />);
  it('expect to render CounterButton component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('correctly incrememts the number onClick', () => {
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({ count: 2 });
    expect(wrapper.props().color).toEqual('purple');
  });
});
