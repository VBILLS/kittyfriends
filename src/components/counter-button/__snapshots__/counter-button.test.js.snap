// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton component expect to render CounterButton component 1`] = `
<button
  color="purple"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
